lib/ir/type.cc

```diff
@@ -201,6 +201,9 @@ block_type* block_type::get(type *elt_ty, const block_shapes_t &shapes) {
   assert(elt_ty && "Can't get a tile of <null> type!");
   assert(shapes.size() && "Can't create a tile with empty shapes!");
   assert(is_valid_elt_ty(elt_ty) && "Invalid type for tile element!");
+  for (size_t i = 0; i < shapes.size(); i++) {
+    assert(shapes[i] % 8 == 0 && "Dimension must be multiply of 8");
```

Comment:
I don't think that this is quite the constraint that Triton blocks need to satisfy. I'd say maybe a reasonable constraint for now is that the number of elements must be (1) a power of two and (2) maybe let's do greater than 32 for now? Not a hard constraint but the compiler is really buggy otherwise